tests/test_spectrum_assignment.py

```diff
@@ -0,0 +1,378 @@
+    nvalmax = oms_list[10].spectrum_bitmap.getn(ind_max) - nguard + 1
+
+    # min index in bitmap must be consistant with min freq attribute
+    print(f'test1 expected: {nvalmin}')
+    print(f'freq_index_min: {oms_list[10].spectrum_bitmap.freq_index_min},')
+    assert nvalmin == oms_list[10].spectrum_bitmap.freq_index_min
+    print(f'test2 expected: {nvalmax}')
+    print(f'freq_index_max: {oms_list[10].spectrum_bitmap.freq_index_max}')
+    assert nvalmax == oms_list[10].spectrum_bitmap.freq_index_max
+    oms_list[10].update_spectrum(freq_min, freq_max, grid=grid, guardband=guardband)
+    # checks that changes are applied on bitmap and freq attributes of Bitmap object
+    print(f'expected: {nmin}, {nmax}')
+    print(f'test3 obtained: {oms_list[10].spectrum_bitmap.freq_index_min},' +\
+          f' {oms_list[10].spectrum_bitmap.freq_index_max}')
+    assert (nmin == oms_list[10].spectrum_bitmap.freq_index_min and
+            nmax == oms_list[10].spectrum_bitmap.freq_index_max)
+
+    print('novel spectrum')
+    print(nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_min) * 1e-12,
+          nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_max) * 1e-12)
+    ind_max = len(oms_list[10].spectrum_bitmap.bitmap) - 1
+    print('with guardband', oms_list[10].spectrum_bitmap.getn(0),
+          oms_list[10].spectrum_bitmap.getn(ind_max))
+    print('without guardband', oms_list[10].spectrum_bitmap.freq_index_min,
+          oms_list[10].spectrum_bitmap.freq_index_max)
+    nvalmin = oms_list[10].spectrum_bitmap.getn(0) + nguard
+    nvalmax = oms_list[10].spectrum_bitmap.getn(ind_max) - nguard + 1
+    # min index in bitmap must be consistant with min freq attribute
+    print(f'expected: {nvalmin}')
+    print(f'freq_index_min: {oms_list[10].spectrum_bitmap.freq_index_min},')
+    print('inconsistancy in Bitmap object')
+    assert nvalmin == oms_list[10].spectrum_bitmap.freq_index_min
+    print(f'expected: {nvalmax}')
+    print(f'freq_index_max: {oms_list[10].spectrum_bitmap.freq_index_max}')
+    print('inconsistancy in Bitmap object')
+    assert nvalmax == oms_list[10].spectrum_bitmap.freq_index_max
+    oms_list = align_grids(oms_list)
+    ind_max = len(oms_list[10].spectrum_bitmap.bitmap) - 1
+    nvalmin = oms_list[10].spectrum_bitmap.getn(0)
+    nvalmax = oms_list[10].spectrum_bitmap.getn(ind_max)
+    print(f'expected: {min(nmin, nvalmin)}, {max(nmax, nvalmax)}')
+    print(f'expected: {nmin, nmax}')
+    print(f'obtained after alignment: {nvalmin}, {nvalmax}')
+    assert nvalmin <= nmin and nvalmax >= nmax
+
+@pytest.mark.parametrize('nval1', [0, 15, 24])
+@pytest.mark.parametrize('nval2', [8, 12])
+def test_assign_and_sum(nval1, nval2, setup):
+    """ checks that bitmap sum gives correct result
+    """
+    network, oms_list = setup
+    grid = 0.00625e12
+    guardband = grid
+    mval = 4 # slot in 12.5GHz
+    freq_min = 193.1e12
+    freq_max = 193.1e12 + 24 * 0.00625e12
```

Comment:
some of these constants (also see comment for the previous test) look like something that can be move to the scope of the whole file